File: 263. Ugly Number/solution.js
/**
 * @param {number} num
 * @return {boolean}
 */
var isUgly = function(num) {
  if (num <= 0) return false;
  for (let p of [2, 3, 5])
    while (num && num % p === 0)
      num /= p;
  return num === 1;
};

File: 263. Ugly Number/solution.go
func isUgly(num int) bool {
	if num <= 0 {
		return false
	}
	if num == 1 {
		return true
	}
	primes := [3]int{5, 3, 2}
	for _, p := range primes {
		for (num % p) == 0 {
			num /= p
		}
	}
	return num == 1
}